### PR TITLE
Set storage s3 bucket cors configuration

### PR DESCRIPTION
**Pull request by yolken-airplane**
## Description
This change updates our agent terraform modules to set cors configurations on any S3 buckets created. This is needed so that uploads work properly from the Airplane FE in the self-hosted storage case.

## Testing
Testing to be completed via applying in our sandbox AWS cluster.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +102,7 @@ module "airplane_agent" {
 | <a name="input_agent_storage_alb_public_subnet_ids"></a> [agent\_storage\_alb\_public\_subnet\_ids](#input\_agent\_storage\_alb\_public\_subnet\_ids) | List of public subnets to use for the external agent ALB. If blank, the agent subnet\_ids will be used instead (which is fine as long as they're public). | `list(string)` | `[]` | no |
 | <a name="input_agent_storage_domain"></a> [agent\_storage\_domain](#input\_agent\_storage\_domain) | For development purposes only | `string` | `"d.airplane.sh"` | no |
 | <a name="input_agent_storage_jwt_public_key"></a> [agent\_storage\_jwt\_public\_key](#input\_agent\_storage\_jwt\_public\_key) | For development purposes only | `string` | `""` | no |
+| <a name="input_agent_storage_s3_allowed_origins"></a> [agent\_storage\_s3\_allowed\_origins](#input\_agent\_storage\_s3\_allowed\_origins) | List of allowed origins to include in agent storage bucket CORS config | `list(string)` | <pre>[<br>  "https://app.airplane.dev"<br>]</pre> | no |
```
Comment (ameyakhare) [inline README.md:105]:
```suggestion
| <a name="input_agent_storage_s3_allowed_origins"></a> [agent\_storage\_s3\_allowed\_origins](#input\_agent\_storage\_s3\_allowed\_origins) | List of allowed origins to include in the agent storage bucket CORS config | `list(string)` | <pre>[<br>  "https://app.airplane.dev"<br>]</pre> | no |
```

File: modules/storage/README.md [hunk 2]
```diff
@@ -48,6 +49,7 @@ See the [Airplane docs](https://docs.airplane.dev/self-hosting/storage) for more
 | <a name="input_team_id"></a> [team\_id](#input\_team\_id) | Airplane team ID - retrieve via `airplane auth info` | `string` | n/a | yes |
 | <a name="input_agent_iam_role_name"></a> [agent\_iam\_role\_name](#input\_agent\_iam\_role\_name) | Name of an existing IAM role to use for the agent. If blank, then a new IAM role will be created. | `string` | `""` | no |
 | <a name="input_agent_storage_domain"></a> [agent\_storage\_domain](#input\_agent\_storage\_domain) | For development purposes only | `string` | `"d.airplane.sh"` | no |
+| <a name="input_agent_storage_s3_allowed_origins"></a> [agent\_storage\_s3\_allowed\_origins](#input\_agent\_storage\_s3\_allowed\_origins) | List of allowed origins to include in agent storage bucket CORS config | `list(string)` | <pre>[<br>  "https://app.airplane.dev"<br>]</pre> | no |
```
Comment (ameyakhare) [inline modules/storage/README.md:52]:
```suggestion
| <a name="input_agent_storage_s3_allowed_origins"></a> [agent\_storage\_s3\_allowed\_origins](#input\_agent\_storage\_s3\_allowed\_origins) | List of allowed origins to include in the agent storage bucket CORS config | `list(string)` | <pre>[<br>  "https://app.airplane.dev"<br>]</pre> | no |
```

File: modules/storage/variables.tf [hunk 3]
```diff
@@ -21,6 +21,12 @@ variable "agent_storage_s3_bucket" {
   default     = ""
 }
 
+variable "agent_storage_s3_allowed_origins" {
+  type        = list(string)
+  description = "List of allowed origins to include in agent storage bucket CORS config"
```
Comment (ameyakhare) [inline modules/storage/variables.tf:26]:
```suggestion
  description = "List of allowed origins to include in the agent storage bucket CORS config"
```

File: variables.tf [hunk 4]
```diff
@@ -65,6 +65,12 @@ variable "agent_storage_s3_bucket" {
   default     = ""
 }
 
+variable "agent_storage_s3_allowed_origins" {
+  type        = list(string)
+  description = "List of allowed origins to include in agent storage bucket CORS config"
```
Comment (ameyakhare) [inline variables.tf:70]:
```suggestion
  description = "List of allowed origins to include in the agent storage bucket CORS config"
```

**Review by ameyakhare — APPROVED** [review]

File: modules/storage/variables.tf (hunk 3 above)
Comment (yolken-airplane) [reply modules/storage/variables.tf:26]: Fixed

File: modules/storage/README.md (hunk 2 above)
Comment (yolken-airplane) [reply modules/storage/README.md:52]: Fixed.

File: variables.tf (hunk 4 above)
Comment (yolken-airplane) [reply variables.tf]: Fixed.

**Pull request merged**